### PR TITLE
refactor(suite): remove unnecessary async from actions

### PR DESCRIPTION
**Pull request by szymonlesisz**
## This PR should be squashed
I did multiple commits to be easy to read, but they should be squashed into one during merge

- unified action names (ScopeAction instead of ScopeActions)
- removed unnecessary redux-thunk or async actions (replaced by regular redux action)
- removed unnecessary types/index files (walletSettings/firmware/onboarding)

### CODE REVIEW

File: packages/suite/src/actions/suite/logActions.ts [hunk 1]
```diff
@@ -5,11 +5,14 @@ import * as Sentry from '@sentry/browser';
...
+export type LogAction =
     | { type: typeof LOG.ADD; payload: LogEntry }
     | { type: typeof LOG.TOGGLE_EXCLUDE_BALANCE_RELATED };
 
-export const addCustom = (action: Action, payload: Record<string, unknown> | undefined): Action => {
+export const addCustom = (
+    action: Action,
+    payload: Record<string, unknown> | undefined,
+): LogAction => {
     return {
```
Comment (szymonlesisz) [inline packages/suite/src/actions/suite/logActions.ts]: => ({ action })

File: packages/suite/src/actions/wallet/discoveryActions.ts [hunk 2]
```diff
@@ -490,7 +490,7 @@ export const start = () => async (dispatch: Dispatch, getState: GetState): Promi
     }
 };
 
-export const stop = () => async (dispatch: Dispatch): Promise<void> => {
+export const stop = () => async (dispatch: Dispatch) => {
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/discoveryActions.ts]: rm async

File: packages/suite/src/actions/suite/metadataActions.ts [hunk 3]
```diff
@@ -680,9 +678,7 @@ export const init = (force = false) => async (dispatch: Dispatch, getState: GetS
     return true;
 };
 
-export const setEditing = (payload: string | undefined) => (dispatch: Dispatch) => {
-    dispatch({
-        type: METADATA.SET_EDITING,
-        payload,
-    });
-};
+export const setEditing = (payload: string | undefined): MetadataAction => ({
```
Comment (szymonlesisz) [inline packages/suite/src/actions/suite/metadataActions.ts:681]: payload?: string

Comment (mroz22) [reply packages/suite/src/actions/suite/metadataActions.ts:681]: depends on taste. If there is a single method that "adds" "removes" something depending on whether argument is provided or not, I personally prefer to type it as required param with union `something | undefined`

Comment (szymonlesisz) [reply packages/suite/src/actions/suite/metadataActions.ts:681]: good :) it's your playground, i'll leave it as is

**Review by mroz22 — APPROVED** [review]
Looks good, should not break anything I believe. What about those 2 comments?

File: packages/suite/src/actions/wallet/discoveryActions.ts (hunk 2 above)
Comment (szymonlesisz) [reply packages/suite/src/actions/wallet/discoveryActions.ts]: cfb49ea

File: packages/suite/src/actions/suite/logActions.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/suite/src/actions/suite/logActions.ts]: 84ccc85

**Comment by szymonlesisz** [comment]
resolved

**Review by mroz22 — APPROVED** [review]

**Pull request merged**